cmd/schema.go

```diff
@@ -46,38 +46,66 @@ environment variables. The schema flags are:
 
 // SetFlags sets the flags.
 func (cmd *SchemaCmd) SetFlags(f *flag.FlagSet) {
+	f.StringVar(&cmd.source, "source", "", "Flag for specifying source DB, (accepted values are \"PostgreSQL\", \"MySQL\", and \"DynamoDB\")")
+	f.StringVar(&cmd.sourceProfile, "source-profile", "", "Flag for specifying connection profile for source database")
+	f.StringVar(&cmd.target, "target", "Spanner", "Specifies the target DB, defaults to Spanner (accepted values: \"Spanner\")")
+	f.StringVar(&cmd.targetProfile, "target-profile", "", "Flag for specifying connection profile for target database")
 	f.StringVar(&cmd.filePrefix, "prefix", "", "File prefix for generated files")
-	f.StringVar(&cmd.driverName, "driver", "pg_dump", "Flag for specifying source DB or dump files (accepted values are \"pg_dump\", \"postgres\", \"mysqldump\", \"mysql\", and \"dynamodb\")")
-	f.Int64Var(&cmd.schemaSampleSize, "schema-sample-size", int64(100000), "Number of rows to use for inferring schema (only for DynamoDB)")
-	f.StringVar(&cmd.dumpFilePath, "dump-file", "", "Location of dump file to process")
-	f.StringVar(&cmd.targetDb, "target-db", "", "Specifies the target DB, defaults to spanner")
 }
 
 func (cmd *SchemaCmd) Execute(ctx context.Context, f *flag.FlagSet, _ ...interface{}) subcommands.ExitStatus {
-	ioHelper := conversion.NewIOStreams(cmd.driverName, cmd.dumpFilePath)
+	sourceProfile, err := NewSourceProfile(cmd.sourceProfile, cmd.source)
+	if err != nil {
+		log.Fatal(err)
```

Comment:
Due to a series of unfortunate adventures, log.Fatal doesn't actually do anything: log messages are dropped on the floor (see background on SetupLogFile in conversion/conversion.go).
That means that all information about why we failed here is lost, which is pretty confusing for users. They just see an "exit status 1" message.

Suggest switching to panic(err) here like we do in main.go.
Longer term, we can resurrect log, but we'll have to deal with the spam issue noted in SetupLogFile that was the reason we initially decided to silence log messages.

Reply:
Thanks for this background, I have switched to panic for now everywhere I used log.Fatal previously.